withReduxSaga/index.js

```diff
@@ -0,0 +1,45 @@
+import createSagaMiddleware, { END } from 'redux-saga';
+import withSaga from './withReduxSaga';
+
+export default (config, InnerComponent, actions) => {
+  const middlwareOptions = config.sagaMonitor ? { sagaMonitor: config.sagaMonitor } : {};
+  const sagaMiddleware = createSagaMiddleware(middlwareOptions);
+
+  const configureStore = config.configureStore(sagaMiddleware);
+
+  const STORE_KEY = '_NEXT_STORE_KEY_';
+  const existingStore = (global.window && global.window[STORE_KEY]);
+  const params = {
+    store: existingStore || {},
+  };
+  params.makeStore = {
+    createStore: (initialState, options) => {
+      params.store = configureStore(initialState, options);
```

Comment:
I made some emphasis on that because i tend to use a file where i build all my initial state.
```js
export default {
  entities: {
    exercises: {},
    frequencies: {},
    goals: {},
    providers: {},
    recipients: {},
    messages: {},
    conversations: {},
  },
  exercises: {
    todaysExercises: {
      isFetching: false,
      list: [],
      error: '',
    },
    library: {
      isFetching: false,
      list: [],
      error: '',
    },
  },
  goals: {
    isFetching: false,
    list: [],
    error: '',
  },
  conversations: {
    isFetching: false,
    isCreating: false,
    lastFetched: '',
    list: [],
    error: '',
    conversation: {
      id: '',
      isFetching: false,
    },
  },
  providers: {
    isFetching: false,
    lastFetched: '',
    list: [],
    error: '',
  },
};
```